src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```
@@ -0,0 +1,51 @@
+package mate.academy.service.impl;
+
+import java.util.ArrayList;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket(movieSession, user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCartFromDb = shoppingCartDao.getByUser(user).get();
+        shoppingCartFromDb.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCartFromDb);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).get();

```

let's throw an exception in case we can't find shopping cart for user